Investigate MLflow models test_cli timeouts when buildi…

andychow-db opened the pull request:
## What changes are proposed in this pull request?

There are three major changes proposed in this pull request.
1. Re-add CLI docker build tests to "large"
2. Updated Ubuntu from 16.04 to 18.04
3. Removed duplicate run of `test/models` in run-large-python-tests.sh`

## How is this patch tested?
Integration tests were run several times because the tests would previously  transiently fail. So far, no run as failed yet.  

Secondly, I temporarily added timing information to `run-large-python-tests.sh` and demonstrated that docker requires approximately 5 minutes to build and the additional tests increase the total time by 5 minutes. In total, the large tests takes approximately 40 to 45 minutes now. 

## Release Notes

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [X] Yes. Give a description of this change to be included in the release notes for MLflow users.

The MLflow CLI will build a Docker image using the ubuntu:18.04 rather than ubuntu:16.04.

### What component(s) does this PR affect?

- [ ] UI
- [X ] CLI
- [ ] API
- [ ] REST-API
- [ ] Examples
- [ ] Docs
- [ ] Tracking
- [ ] Projects
- [ ] Artifacts
- [ ] Models
- [ ] Scoring
- [ ] Serving
- [ ] R
- [ ] Java
- [ ] Python

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [X ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

andychow-db committed:
[ML-8353] Investigate OSS MLflow models test_cli timeouts when building docker images

andychow-db committed:
Add timing command to run-large-python-tests.sh

andychow-db committed:
Update docker ubuntu to 18.04

andychow-db committed:
Add test/model to ignored

andychow-db committed:
Revert "Update docker ubuntu to 18.04"

This reverts commit c9d5c7ea81bdf437935fc55658369ccbafde6be2.

andychow-db committed:
Removing timing

andychow-db committed:
Revert "Revert "Update docker ubuntu to 18.04""

This reverts commit e06dd984680c462a18da7495a2c742991ddb6085.

andychow-db committed:
Readd test_spark_model_export docker based tests

dbczumar reviewed (approved):
LGTM! Thanks @andychow-db

andychow-db committed:
Fix test_prepare_env_fails

The test is designed to validate that prepare_env will fail when the conda env file has a problematic parameter
The newest version of conda smartly ignores problematic parameters and successfully runs the command.

dbczumar commented on tests/models/test_cli.py:
```diff
@@ -278,21 +278,20 @@ def test_prepare_env_fails(sk_model):
     with TempDir(chdr=True):
         with mlflow.start_run() as active_run:
-            mlflow.sklearn.log_model(sk_model, "model", conda_env={"env": "Bad conda env"})
+            mlflow.sklearn.log_model(sk_model, "model", conda_env={"dependencies": ["mlflow-does-not-exist-dep==abc"]})
```
LGTM

andychow-db committed:
Fix linting
